libs/auth/src/openssl.ts

```diff
@@ -218,11 +218,24 @@ export async function verifySignature({
     '-verify',
     publicKey,
     '-signature',
-    challengeSignature,
-    challenge,
+    messageSignature,
+    message,
   ]);
 }
 
+/**
+ * Signs a message using a private key file (in PEM format)
+ */
+export async function signMessageUsingPrivateKeyFile({
```

Comment:
This is just for dev. In prod, we'll be using the TPM and a variant of [this script](https://github.com/votingworks/vxsuite-complete-system/blob/a6f272bc74684807c87d80be06926550fe61ab4a/app-scripts/sign.sh)

Reply:
Good q! It's already enforced in the sense that we have no private key files in prod, just the private key in the TPM, which can't be directly accessed.

I considered a `NODE_ENV` check, but there's some nuance there since we have some prod-like environments (VxDev, integration tests) where this function is okay to use.

All this said, I do think a comment makes sense! And maybe some code tweaks - let me think some more about this

Reply:
Okay, thought some more about this and restructured this helper to support both the dev and prod (i.e. file key and TPM key) use cases: https://github.com/votingworks/vxsuite/pull/3476/commits/8344fdc081601a8c5da15049ddc37db713e78e09

I'd previously not done this because I wasn't planning on using OpenSSL for signing messages with the TPM key (I was planning on using [this script](https://github.com/votingworks/vxsuite-complete-system/blob/a6f272bc74684807c87d80be06926550fe61ab4a/app-scripts/sign.sh) instead). But I just successfully used OpenSSL to sign with the TPM key on my ThinkPad and am now leaning toward that approach for consistency with cert creation functions and consistency in general.

I haven't actually added the code for signing messages with the TPM key via OpenSSL to this PR since I think that merits a separate review.